src/commands/mod.rs

```diff
@@ -289,6 +293,69 @@ impl DaemonControl {
         GetAddressResult::new(address)
     }
 
+    /// list addresses
+    pub fn list_addresses(
+        &self,
+        start_index: Option<u32>,
+        count: Option<u32>,
+    ) -> Result<ListAddressesResult, CommandError> {
+        let mut db_conn = self.db.connection();
+        let receive_index: u32 = db_conn.receive_index().into();
+        let change_index: u32 = db_conn.change_index().into();
+
+        let start_index = start_index.unwrap_or(0);
+
+        if start_index > (2u32.pow(31) - 1) {
+            return Err(CommandError::InvalidAddressIndex);
+        }
+
+        let count = count.unwrap_or_else(|| receive_index.max(change_index) - start_index);
+
+        if count == 0 {
+            let out: Vec<AddressInfo> = Vec::new();
+            return Ok(ListAddressesResult::new(out));
+        }
```

Comment:
It's a bit ugly to special case 0 here to avoid the `checked_sub` below to return `None`. Could get rid of the necessity for a checked_sub altogether by using an exclusive range (`(x..y)`) below instead of an inclusive one (`(x..=y)`).